test/manual-test-examples/webgl/geometry/2DPrimitives/framesketch.js

```diff
@@ -0,0 +1,265 @@
+    triangle(-38, -51, 86, -20, 69, 63);
+    triangle(38, 51, -69, -63, -86, 20);
+  }
+
+ if(showThisTest()){
+    testinfo.html('Line, animated position, width, color. Animated color was included in case stroke '+
+                  'implementation in WebGL does not work correctly');
+    strokeWeight(floor(Math.cos(millis()/1000) * 20)+21);
+    stroke(255, floor(millis()/10%255), floor((millis()-100)/10%255));
+    line(-150,-150,100 + Math.cos(millis()/1000) * 50
+                  ,100 + Math.sin(millis()/1000) * 50);
+  }
+
+
+  if(showThisTest()){
+    testinfo.html('Reference line, and lines with stroke caps specified');
+    strokeWeight(20);
+    stroke(255,100,100);
+    line(-80, -50, 80, -50);
+    stroke(100,200,255);
+    strokeCap(ROUND);
+    line(-80, 0, 80, 0);
+    strokeCap(SQUARE);
+    line(-80, 50, 80, 50);
+    strokeCap(PROJECT);
+    line(-80, 90, 80, 90);
+  }
+
+  if(showThisTest()){
+    testinfo.html('Shape with contours');
+    stroke(255, 0, 0);
+    beginShape();
+    // Exterior part of shape, clockwise winding
+    vertex(-40, -40);
+    vertex(40, -40);
+    vertex(40, 40);
+    vertex(-40, 40);
+    // Interior part of shape, counter-clockwise winding
+    beginContour();
+    vertex(-20, -20);
+    vertex(-20, 20);
+    vertex(20, 20);
+    vertex(20, -20);
+    endContour();
+    endShape(CLOSE);
+  }
+
+  if(showThisTest()){
+    testinfo.html('Shape with quadraticVertex');
+    noFill();
+    strokeWeight(4);
+    beginShape();
+    vertex(20, 20);
+    quadraticVertex(80, 20, 50, 50);
+    quadraticVertex(20, 80, 80, 80);
+    vertex(80, 60);
+    endShape();
+  }
+
+  if(showThisTest()){
+    testinfo.html('Shapes drawn with vertex. with bowtie shapes');
+    stroke(255, 0, 0);
+    beginShape();
+    // Exterior part of shape, clockwise winding
+    vertex(-40, -40);
+    vertex(40, -40);
+    vertex(40, 40);
+    vertex(-40, 40);
+    endShape(CLOSE);
+
+    beginShape();
+    // Exterior part of shape, clockwise winding
+    vertex(-140, -140);
+    vertex(-60, -140);
+    vertex(-60, -60);
+    vertex(-40, -80);
+    endShape(CLOSE);
+    
+    beginShape();
+    // Exterior part of shape, clockwise winding
+    vertex(140, -140);
+    vertex(60, -140);
+    vertex(60, -60);
+    vertex(40, -80);
+    endShape(CLOSE);
+  }
+
+  if(showThisTest()){
+    testinfo.html('Points, with size and color animation');
+    strokeWeight(floor(Math.cos(millis()/1000) * 40)+1);
+    stroke(255, floor(millis()/10%255), floor((millis()-100)/10%255));
+    point(-20, -20);
+    point(20, 40);
+  }
+
+  if(showThisTest()){
+    testinfo.html('Smooth test (no smooth is on the bottom right)')
+    background(0);
+    noStroke();
+    smooth();
+    ellipse(-100, -100, 200, 200);
+    noSmooth();
+    ellipse( 100,  100, 200, 200);
+  }
+}
+
+function showThisTest(){
+  //Helper function so the tests can be reordered easily.
+  n++
```

Comment:
this is clever but I think it would be good if you could provide a deeper comment at the beginning of draw, that showThisTest will increment the number n and test if that matches the current test parameter, resulting in only one test being run but checking them all. it's a little confusing that all the conditionals are the same...